### PR TITLE
Embedded Sandbox Relay

### PR DESCRIPTION
**Pull request by mayakoneval**
## Context

We are exposing and embedded Sandbox! Yay. This PR is the parent page interaction with the embed. That includes making requests, making introspection requests, rendering the iframe with the correct query params. 

## What changed

This PR implements
1. Allowing the sandbox path to be embedded on `https://sandbox.embed.apollographql.com/sandbox`
2. Allowing for a post messaged introspection schema provider, which sends post messages to the parent at a regular poll interval, and builds the schema for the schema pages from that.
3. passing through the current endpoint for each EXPLORER_REQUEST so the updated endpoint is queried

## How to test this PR
1. checkout branch in `studio-ui` repo
2. npm run start:embedded
3. checkout [corresponding branch](https://github.com/apollographql/embeddable-explorer/pull/80) in `embeddable-explorer` repo
4. switch the `EMBEDDABLE_EXPLORER_URL = 'https://embed.apollo.local:3000/sandbox'`;
5. `npm run build-sandbox:umd` in `embeddable-explorer` repo. 
6. Open `localDevelopmentExample` from sandbox examples. 

## Questions
1. Should we switch the url to `https://sandbox.embed.apollographql.com/sandbox` so all the links still work, but the subdomain is slightly different?
2. should sandbox local storage in embeds carry over to all other sandboxes or should embed local storage be namespaced by origin?

## Still TODO
1. I need to implement all the outlinks from Sandbox. Right now none of them work. That will come in a follow up PR rebased off of [william/explorer-everywhere/authorize-embed](https://github.com/mdg-private/studio-ui/tree/william/explorer-everywhere/authorize-embed)
3. Auth is not complete, so no one can login

### CODE REVIEW

File: src/embeddedExplorer/examples/sandboxLocalDev.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>Embedded Explorer Example</title>
```
Comment (William010x) [inline src/embeddedExplorer/examples/sandboxLocalDev.html:5]: nbd: we can probably rename this Embedded Sandbox Example

File: src/embeddedExplorer/examples/sandboxLocalDev.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      {
+        // Note that the target div must be in the dom before the script tag creating the embedded explorer
+      }
+    </script>
+    <div
+      style="width: 800px; height: 700px; margin: auto; border: none"
+      id="embeddableSandbox"
+    ></div>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <script src="../../dist/embeddable-explorer.umd.production.min.js"></script>
```
Comment (William010x) [inline src/embeddedExplorer/examples/sandboxLocalDev.html:26]: I'm running into `Uncaught TypeError: window.EmbeddedSandbox is not a constructor` error when I open sandboxLocalDev. Should this be `../../../dist/embeddable-sandbox.umd.production.min.js`?

**Review by William010x — APPROVED** [review]
Looks good! This is really cool 🎉

File: src/helpers/constants.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// URL for any embedded Explorer iframe
+export const EMBEDDABLE_EXPLORER_URL =
+  'https://explorer.embed.apollographql.com';
+export const EMBEDDABLE_SANDBOX_URL =
+  'https://embed.apollo.local:3000/sandbox/explorer';
```
[the remark this answers is not in the document]
Comment (mayakoneval) [reply src/helpers/constants.ts]: yes :) whoops!

File: src/embeddedExplorer/examples/sandboxLocalDev.html (hunk 2 above)
Comment (mayakoneval) [reply src/embeddedExplorer/examples/sandboxLocalDev.html]: this whole file was actually an old development iteration, I'll delete it!

File: .circleci/config.yml [hunk 4]
```diff
@@ -133,25 +133,100 @@ jobs:
...
+      - gcp-cli/initialize:
+          gcloud-service-key: GCLOUD_GCS_SERVICE_KEY
+      - gcp-cli/install
+      - checkout
+      - attach_workspace:
+          at: ~/embeddable-explorer
+      - run:
+          name: Upload build artifacts to CDN
+          command: |
+            gsutil -m rsync -r ~/embeddable-sandbox/dist gs://embeddable-sandbox/_latest
```
Comment (mayakoneval) [inline .circleci/config.yml:178]: @cheapsteak I created a new bucket for these files: https://console.cloud.google.com/storage/browser/embeddable-sandbox;tab=objects?project=mdg-services&prefix=&forceOnObjectsSortingFiltering=false

Just an fyi

**Pull request merged**